modin/experimental/core/storage_formats/hdk/query_compiler.py

```diff
@@ -557,18 +557,27 @@ def concat(self, axis, other, **kwargs):
 
     def drop(self, index=None, columns=None, errors: str = "raise"):
         if index is not None:
+            # Only column drop is supported by the HDK engine
             raise NotImplementedError("Row drop")
         if errors != "raise":
             raise NotImplementedError(
                 "This lazy query compiler will always "
                 + "raise an error on invalid columns."
             )
-        return self.__constructor__(
-            self._modin_frame.take_2d_labels_or_positional(
-                row_labels=index, col_labels=self.columns.drop(columns)
-            )
+
+        columns = self.columns.drop(columns)
+        new_frame = self._modin_frame.take_2d_labels_or_positional(
+            row_labels=index, col_labels=columns
         )
 
+        # If all columns are dropped and the index is trivial, we are
+        # not able to restore it, since we don't know the number of rows.
+        # In this case, we copy the index from the current frame.
+        if len(columns) == 0 and new_frame._index_cols is None:
+            new_frame._index_cache = self._modin_frame.index.copy()
```

Comment:
let's add this assertion just to be sure we're safe here
```suggestion
        if len(columns) == 0 and new_frame._index_cols is None:
            assert index is None, "Can't copy old indexes as there was a row drop"
            new_frame._index_cache = self._modin_frame.index.copy()
```

Reply:
Let's use `ValueError` exception, it's more reliable.

Reply:
Yes, I know that. The `assert` is just another safety measure for the case we remove the check on 559 line someday.